pkg/catalog/server.go

```diff
@@ -19,6 +20,7 @@ func (p *Generator) Update(m *sysl.Module, errs ...error) *Generator {
 	//p.Fs = afero.NewMemMapFs()
 	p.RootModule = m
 	p.GeneratedFiles = make(map[string][]byte)
+	p.GeneratedFilesMutex = &sync.Mutex{}
```

Comment:
Just curious, was there any reason why sync.Mutex is used here instead of sync.RWMutex like above since it's applying to a map still, so it's safe for concurrent reads?